crml/transaction-payment/src/lib.rs

```diff
@@ -325,15 +322,264 @@ where
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use crate::mock::*;
-	use frame_support::weights::DispatchClass;
-	use sp_runtime::{testing::TestXt, traits::Extrinsic};
+	use codec::Encode;
+	use frame_support::{
+		impl_outer_dispatch, impl_outer_origin, parameter_types,
+		weights::{DispatchClass, DispatchInfo, GetDispatchInfo, Weight},
+	};
+	use pallet_balances::Call as BalancesCall;
+	use pallet_transaction_payment_rpc_runtime_api::RuntimeDispatchInfo;
+	use sp_core::H256;
+	use sp_runtime::{
+		testing::{Header, TestXt},
+		traits::{BlakeTwo256, Extrinsic, IdentityLookup},
+		Perbill,
+	};
+	use std::cell::RefCell;
+
+	use crate::{Module, Trait};
+	use cennznet_primitives::{
+		traits::{BuyFeeAsset, IsGasMeteredCall},
+		types::FeeExchange,
+	};
+	use frame_support::{
+		additional_traits::DummyDispatchVerifier,
+		dispatch::DispatchError,
+		traits::{Currency, Get},
+	};
+	use sp_runtime::traits::Convert;
+
+	const VALID_ASSET_TO_BUY_FEE: u32 = 1;
+	const INVALID_ASSET_TO_BUY_FEE: u32 = 2;
+	// Transfers into this account signal the extrinsic call should be considered gas metered
+	const GAS_METERED_ACCOUNT_ID: u64 = 10;
 
 	// A balance transfer
-	const CALL: &<Runtime as frame_system::Trait>::Call = &mock::Call::Balances(pallet_balances::Call::transfer(2, 69));
+	const CALL: &<Runtime as frame_system::Trait>::Call = &Call::Balances(BalancesCall::transfer(2, 69));
+
 	// A balance transfer, which will be considered 'gas metered' for testing purposes
 	const METERED_CALL: &<Runtime as frame_system::Trait>::Call =
-		&mock::Call::Balances(pallet_balances::Call::transfer_keep_alive(GAS_METERED_ACCOUNT_ID, 69));
+		&Call::Balances(BalancesCall::transfer_keep_alive(GAS_METERED_ACCOUNT_ID, 69));
+
+	impl_outer_dispatch! {
+		pub enum Call for Runtime where origin: Origin {
+			pallet_balances::Balances,
```

Comment:
Was there a reason to move this setup stuff out of mock.rs?

Reply:
The entire `mock.rs` file is removed and the contents are move here

Reply:
Just to be comparable with Plug to make the possible future migration easier.